### PR TITLE
[Handshake] Improve handshake builders

### PR DESCRIPTION
**Pull request by Dinistro**
These are two very small changes and I would commit such things directly (when the CI is fine with them). 
I just wanted to ask if it is preferred to keep these different changes in separate commits or if I should squash them.

### CODE REVIEW

**Comment by mortbopet** [comment]
I'd say split it - the `null` checks on `op` are trivial, but the change to the `handshake.func` parser should have an additional test where the visibility keyword is used.

File: lib/Dialect/Handshake/HandshakeOps.cpp [hunk 1]
```diff
@@ -228,10 +229,11 @@ void LazyForkOp::build(OpBuilder &builder, OperationState &result,
   // Fork is control-only if it is the no-data output of a ControlMerge or a
   // StartOp
   auto *op = operand.getDefiningOp();
-  bool isControl = ((dyn_cast<ControlMergeOp>(op) || dyn_cast<StartOp>(op)) &&
-                    operand == op->getResult(0))
-                       ? true
-                       : false;
+  bool isControl =
+      op && ((dyn_cast<ControlMergeOp>(op) || dyn_cast<StartOp>(op)) &&
```
Comment (uenoku) [inline lib/Dialect/Handshake/HandshakeOps.cpp:233]: nit: `isa_and_nonnull<ControlMergeOp, StartOp>(op)` would be cleaner.

File: lib/Dialect/Handshake/HandshakeOps.cpp [hunk 2]
```diff
@@ -228,10 +228,12 @@ void LazyForkOp::build(OpBuilder &builder, OperationState &result,
   // Fork is control-only if it is the no-data output of a ControlMerge or a
   // StartOp
   auto *op = operand.getDefiningOp();
-  bool isControl = ((dyn_cast<ControlMergeOp>(op) || dyn_cast<StartOp>(op)) &&
-                    operand == op->getResult(0))
-                       ? true
-                       : false;
+  bool isControl = isa_and_nonnull<ControlMergeOp, StartOp>(op) &&
```
Comment (cgyurgyik) [inline lib/Dialect/Handshake/HandshakeOps.cpp:231]: Why not place this in a function rather than duplicating it 3 times?

Comment (Dinistro) [reply lib/Dialect/Handshake/HandshakeOps.cpp:231]: That might be a good idea, I agree. Should I add this or do we keep these changes for later?

Comment (cgyurgyik) [reply lib/Dialect/Handshake/HandshakeOps.cpp:231]: Just a flyby suggestion, since I saw the same expression 3 times. Up to you :-).

**Pull request merged**